### ISSUE TITLE
Survey runner does not follow `--out` flag

### ISSUE THREAD

**Issue by nicholascapo** [saltstack/salt#24460]
The survey runner (both `.hash` and `.diff`) displays results with printed strings that do not follow the `--out` flag [1] [2]. This makes it very hard to parse the output.

JSON Example:

```bash
$ sudo salt-run --out=json survey.hash '*' grains.get num_cpus
```

Expected [3]:

```json
[
  {
    "result": "2",
    "pool": [
      "host1"
    ]
  },
  {
    "result": "4",
    "pool": [
      "host2"
    ]
  }
]
```

Got (markers added):

```
minion pool :   # <=============== not valid json
------------
['host1']
pool size :
----------
    1
pool result :
-------
    2


minion pool :
------------
['host2']
pool size :
----------
    1
pool result :
-------
    4


[               # <=============== valid json starts here
    {
        "result": "2", 
        "pool": [
            "host1"
        ]
    }, 
    {
        "result": "4", 
        "pool": [
            "host2"
        ]
    }
]

```

[1] https://github.com/saltstack/salt/blob/develop/salt/runners/survey.py#L54

[2] https://github.com/saltstack/salt/blob/develop/salt/runners/survey.py#L106

[3] The structure here is not a requirement, just an example of what I expected from `--out=json`. In this case I am simply omitting the duplicate data that is printed.

**Comment by rallytime** [comment]
Thanks for the report @nicholascapo. Are you using the develop branch or what version of salt are you running `salt --versions-report`?

**Comment by nicholascapo** [comment]
@rallytime Sorry forgot that part.

```
$ salt-call --versions-report
                  Salt: 2015.5.0
                Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
                Jinja2: 2.6
              M2Crypto: 0.21.1
        msgpack-python: 0.1.10
          msgpack-pure: Not Installed
              pycrypto: 2.6
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 13.1.0
                  RAET: Not Installed
                   ZMQ: 3.2.3
                  Mako: 0.7.0
 Debian source package: 2015.5.0+ds-1~bpo70+1

```

**Comment by nicholascapo** [comment]
The only part of this which is still broken is survey.diff, I am not sure how to handle the additional output, since it is useful information.
